docs-ref-conceptual/format-output-azure-cli.md

```diff
@@ -181,44 +186,117 @@ The next example shows how `tsv` output can be piped to other commands in bash.
+- keys
+
+To avoid command output being written to your log, use one of these options:
+
+|Option|Benefit|Use case|
+|-|-|-|
+|`--output none` output format| Keeps security information from displaying in your console _and from being written to your log_. If your command fails, you'll still receive error messages.| 1. Use when command output _can be_ retrieved at a later time.|
+| | | 2. Use when you have no need for output.
+| | | 3. A common choice when a managed identity or a service principal is being used to manage Azure resources.
+|`--query` parameter | Stores output in a variable. Doesn't store command output in the log.|1. Use when command output _can't be_ retrieved at a later time.|
+| | | 2. Use when you need to use a command output value in a script.
+
+### Use `none` and retrieve security information at a later time
+
+_Some_ Azure secrets can be retrieved at a later time. A good example is secrets stored in Azure Key Vault. In this example, create an Azure Key Vault secret using [az keyvault secret set](/cli/azure/keyvault/secret#az-keyvault-secret-set) with the `--output none` option. You can retrieve the secret later using the [az keyvault secret show](/cli/azure/keyvault/secret#az-keyvault-secret-show) command.
+
+```azurecli-interactive
+az keyvault secret set --name MySecretName \
+                       --vault-name MyKeyVaultName \
+                       --value MySecretValue\
+                       --output none
 ```
 
+### Use `--query` and return security information to a variable
+
+The use of `--query` to store output in a variable is technically not an output format. It is a solution to protect secrets, and is an alternative to using `--output none`. For example, when you reset a service principal credential, the password can't be retrieved again.
+
+Reset a service principal credential returning output in the default json format:
+
+```azurecli-interactive
+# reset service principal credentials returning results to the console and the log
+az ad sp credential reset --id myServicePrincipalID --output json
+```
+
+Console output showing the new password in the console. This information is also written in the log.
+
 ```output
-Welcome to the Azure CLI! This command will guide you through logging in and setting some default values.
+{
+  "appId": "myServicePrincipalID",
+  "password": "myServicePrincipalNewPassword",
+  "tenant": "myTenantID"
+}
+```
 
-Your settings can be found at /home/defaultuser/.azure/config
-Your current configuration is as follows:
+A better solution is to return security information to a variable. This example _doesn't_ write the service principal password to the log. When testing, use the `echo` command to see the value of your variable, but understand that `echo` writes to the log.
 
-  ...
+```azurecli-interactive
+# reset service principal credentials returning results to a variable
+myNewPassword=$(az ad sp credential reset --id myServicePrincipalID --query password --output tsv)
+```
```

Comment:
I think it would be better if we call out this is a Bash script. It doesn't work in PowerShell or CMD.